hachoir/metadata/main.py

```diff
@@ -144,6 +162,59 @@ def profile(values, filenames):
     return runProfiler(processFiles, (values, filenames), {'display': False})
 
 
+def getMetadata(args, display=False):
+    """
+    Return a list of dictionaries.
+
+    Take a list of argument(s) and file path(s).
+    Must have a file path in the list.
+
+    Examples:
+        getMetadata(["--raw", "C:\\myVid.mkv"])
+        getMetadata(["--raw", "C:\\myVid.mkv", "D\\myPic.jpg", "--mime"])
+
+    Arguments available for the list:
+        --type      Only display file type (description) - default is False
+        --mime      Only display MIME type - default - default is False
+        --level     Quantity of information to display from 1 to 9 (9 is the maximum) - default is 9
+        --raw       Raw output - default is False
+        --quality   Information quality (0.0=fastest, 1.0=best, and default is 0.5)
+        --maxlen    Maximum string length in characters, 0 means unlimited
+                        (default is defined in ...\\hachoir\\metadata\\config.py (usually 300))
+
+    Raises:
+        FileNotFoundError: If the args list do not contain any file path.
+        ValueError: If the args list contain an not supported arg.
+    """
+    valid_options = [
+        "--type",
+        "--mime",
+        "--level",
+        "--raw",
+        "--quality",
+        "--maxlen",
+    ]
+
+    # Validate args
+    file_path = [arg for arg in args if os.path.exists(arg)]
+    if not file_path:
+        raise FileNotFoundError("No file path in args")
+    invalid_options = [
+        arg
+        for arg in args
+        if arg not in valid_options and arg not in file_path
+    ]
+    if invalid_options:
+        raise ValueError("Invalid arguments")
+
+    # Parser options and initialize Hachoir
+    values, filenames = parseOptions(args)
+
+    config.RESULT_DICTS = []
```

Comment:
It is really surprising to store results in a global variable in the config module. Why not modifying the function to return a list instead? I would prefer to avoid such surprising side effects.